File: playground/python/app/agent_studio.py
from os import environ
from asyncio import run

from algoliasearch.agent_studio.client import AgentStudioClient
from dotenv import load_dotenv

load_dotenv("../.env")


async def main():
    client = AgentStudioClient(
        environ.get("ALGOLIA_APPLICATION_ID"), environ.get("ALGOLIA_ADMIN_KEY"), "us"
    )

    print("client initialized", client)

    try:
        response = await client.list_agents()

        print("List of agents:")
        for agent in response.data:
            print(f"- {agent.name} (ID: {agent.id})")
    finally:
        await client.close()

        print("client closed", client)


run(main())

File: playground/scala/src/main/scala/AgentStudio.scala
import algoliasearch.api.AgentStudioClient
import algoliasearch.agentstudio.JsonSupport
import io.github.cdimascio.dotenv.Dotenv
import org.json4s.Formats

import scala.concurrent.duration.Duration
import scala.concurrent.{Await, ExecutionContextExecutor}

object AgentStudio {
  def main(args: Array[String]): Unit = {
    implicit val ec: ExecutionContextExecutor = scala.concurrent.ExecutionContext.global
    implicit val formats: Formats = JsonSupport.format

    val dotenv = Dotenv.configure.directory("../").load
    val appId = dotenv.get("ALGOLIA_APPLICATION_ID")
    val apiKey = dotenv.get("ALGOLIA_ADMIN_KEY")

    val client = AgentStudioClient(
      appId = appId,
      apiKey = apiKey,
      region = "us",
    )

    val res = client.listAgents()
    val response = Await.result(res, Duration(100, "sec"))

    println("List of agents:")
    for (agent <- response.data) {
      println(s"- ${agent.name} (ID: ${agent.id})")
    }
  }
}
